zvariant/src/dict.rs

```diff
@@ -151,6 +154,14 @@ impl<'k, 'v> Dict<'k, 'v> {
         }
     }
 
+    pub fn iter(&self) -> Iter<'_, 'k, 'v> {
+        Iter(self.map.iter())
+    }
+
+    pub fn iter_mut(&mut self) -> IterMut<'_, 'k, 'v> {
```

Comment:
I think best to return `impl Iterator<Item = .. >` so we don't have to do wrap and implement all the traits.

Reply:
> ```rust
> type IntoIter = impl Iterator<Item = Self::Item>;
> ```

Hmm.. but we do you need this to be generic? Wouldn't this just be `BTreeMap::IntoIterator<Self::Item>`?

Reply:
Yes, that's doable. I've updated the logics in the second commit.
Note: I'll squash the commits into one after all reviews are finished and turn the PR into ready.